website/src/auth.tsx

```diff
@@ -64,6 +68,54 @@ export const UserProvider = (props: { children: any }) => {
     return
   }, [user])
 
+  function createUser(username: string, password: string) {
+    console.log(`requesting adding user ${username} to Cognito User Pool`)
+    let userAttributes = [{ Name: "email", Value: username }].map((attr) => {
+      return new CognitoUserAttribute(attr)
+    })
+    userPool.signUp(
+      username,
+      password,
+      userAttributes,
+      [],
+      async (err, result) => {
+        if (err) {
+          alert(err.message || JSON.stringify(err))
+          return
+        }
+        let cognitoUser = result?.user
+        if (!cognitoUser) {
+          console.log("failed to create user and activate session")
+          return
+        }
+        setUser(cognitoUser)
+        await navigate("/auth/confirm-signup")
+      }
+    )
+  }
+
+  function resetVerificationCode() {
+    user?.resendConfirmationCode((err, result) => {
+      if (err) {
+        alert(err.message || JSON.stringify(err))
+        return
+      }
+      console.log(result)
+    })
+  }
+
+  function verifyUser(verificationCode: string) {
+    user?.confirmRegistration(verificationCode, false, (err, result) => {
+      if (err) {
+        alert(err.message || JSON.stringify(err))
+        return
+      }
+      console.log(result)
+      setUser(user)
+      navigate("/auth/login") 
```

Comment:
Does the user have to sign in again after verifying? If so, why do we call `setUser` above here?